src/tests/integration/components/myGallery.test.js

```diff
@@ -0,0 +1,159 @@
+import { mount } from '@vue/test-utils'
+import { describe, it, expect, vi, beforeEach } from 'vitest'
+import { createPinia, setActivePinia, defineStore } from 'pinia'
+import flushPromises from 'flush-promises'
+import MyGallery from '../../../components/Images/MyGallery.vue'
+import { fetchApiCall } from '../../../utils/api.js'
+import { formatDate } from '../../../utils/formatTime.js'
+
+// Mock the fetchApiCall function
+vi.mock('../../../utils/api.js', () => ({
+  fetchApiCall: vi.fn()
+}))
+
+// Creates a fresh component instance so that each test is isolated
+// Avoids cross-test pollution and ensures a clean slate for each test
+function createComponent () {
+  const pinia = createPinia()
+  setActivePinia(pinia)
+
+  const useSessionsStore = defineStore('sessions', {
+    state: () => ({
+      sessions: {
+        results: [
+          { id: 'session1', start: '2024-08-01T12:00:00Z' },
+          { id: 'session2', start: '2024-08-01T12:30:00Z' }
+        ]
+      }
+    })
+  })
+
+  const useUserDataStore = defineStore('userData', {
+    state: () => ({
+      authToken: 'mock-token'
+    })
+  })
+
+  const useConfigurationStore = defineStore('configuration', {
+    state: () => ({
+      thumbnailArchiveUrl: 'http://mock-api.com/'
+    })
+  })
+
+  // Initialize the stores
+  const sessionsStore = useSessionsStore()
+  const userDataStore = useUserDataStore()
+  const configurationStore = useConfigurationStore()
```

Comment:
Once you test more components will this stuff above here migrate to a shared test class that can be used by all the components? It seems like it would be common to most things you need to test.

Reply:
yeah, I'll do that